### PR TITLE
Add quart compatibility.

### PR DESCRIPTION
**Pull request by Ristellise**
This PR adds:  
- `quart_compat.py` [A simple test module with 1 function only.]
- `async_compat.py` [Conditional import]
- Quart compatibility fixes to get it working with it.

If this PR gets accepted, this would fix issues with quart with flask-security.  
This does not affect any flask functionality as far as I'm aware.  
If I need to fix/change/lint anything please let me know.

### CODE REVIEW

File: flask_security/cli.py [hunk 1]
```diff
@@ -78,7 +103,6 @@ def users_create(identity, password, active):
 @roles.command("create")
 @click.argument("name")
 @click.option("-d", "--description", default=None)
-@click.option("-p", "--permissions")
```
Comment (jwag956) [inline flask_security/cli.py]: This is a new capability in Flask-Security.. so shouldn't be removed.

File: flask_security/utils.py [hunk 2]
```diff
@@ -612,7 +612,7 @@ def default_want_json(req):
     if req.is_json:
         return True
     # TODO should this handle json sub-types?
-    if req.accept_mimetypes.best == "application/json":
+    if req.accept_mimetypes["application/json"] >= 1.0:
```
Comment (jwag956) [inline flask_security/utils.py]: Not sure this is quite the same condition - I realize that quart doesn't fully support werkzeug - however if you support http spec - you probably need to support it. In any case - if you can't - lets make this another conditiponal.

File: flask_security/cli.py (hunk 1 above)
Comment (Ristellise) [reply flask_security/cli.py]: I'll fix that. Code was mainly copy pasted from an older source version.

File: flask_security/utils.py (hunk 2 above)
Comment (Ristellise) [reply flask_security/utils.py]: After adding this, I think your right. Problem is that quart doesnt have a `.best` in their function.

We could try to polyfill or make our on method though...

**Comment by Ristellise** [comment]
> Also - I don't see any tests - so that might mean in the future things get broken?  
Probably but the only thing that would break the entire system would be quart_compat.py check. which is unlikely unless they renamed flask_patch to something else.

**Comment by jwag956** [comment]
For now - I am following Pallets lead on keeping py2.7 support. I don't know when they will drop it - hopefully soon!
Until then - I would like to keep it working - so could you add a if quart and PY3: around async code? you don't support 2.7 - so it shouldn't matter.

**Comment by Ristellise** [comment]
> so could you add a if quart and PY3: around async code? you don't support 2.7 - so it shouldn't matter.

Not really. The issue is that `async` keyword causes a syntax error on all python 2 versions...  

On second thoughts doing a try and except *could work...*

**Comment by Ristellise** [comment]
Try and catch doesnt seem to work on python 2.7 tests...

We COULD try this: https://stackoverflow.com/a/57023749 (An exec a hardcoded script source. But at that point, "Is it worth it?"

**Comment by jwag956** [comment]
I cloned your PR, fixed coverage issues and merged...

Thanks!

**Comment by Ristellise** [comment]
Thanks again for all your work. :)

**Pull request closed without merging**